include/nuttx/mutex.h

```diff
@@ -164,7 +197,26 @@ static inline int nxmutex_lock(FAR mutex_t *mutex)
 
 static inline int nxmutex_trylock(FAR mutex_t *mutex)
 {
-  return nxsem_trywait(mutex);
+  int ret;
+
+  for (; ; )
```

Comment:
I think we do not need `for` in `trylock`

Reply:
But the caller doesn't expect to receive -EINTR/-ECANCEL since nxmutext_lock doesn't return them too.

Reply:
I need to check, but if I recall correctly those error codes are not returned by trywait.

Reply:
yes,you are right,sem_trywait not return this errcode

Reply:
If so, let's keep the simple one. Thanks.